pyproject.toml

```diff
@@ -1,48 +1,179 @@
-write_to = "src/vector/version.py"
+[project]
+name = "vector"
+description = "Vector classes and utilities"
+readme = { file = "README.md", content-type = "text/markdown" }
+maintainers = [ {name = "The Scikit-HEP admins", email = "scikit-hep-admins@googlegroups.com"} ]
+license = "BSD-3-Clause"
+requires-python = ">=3.6"
+keywords = [
+    "vector",
+]
+classifiers = [
+    "Development Status :: 5 - Production/Stable",
+    "Intended Audience :: Developers",
+    "Intended Audience :: Science/Research",
+    "License :: OSI Approved :: BSD License",
+    "Operating System :: OS Independent",
+    "Programming Language :: Python",
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3 :: Only",
+    "Programming Language :: Python :: 3.6",
+    "Programming Language :: Python :: 3.7",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Topic :: Scientific/Engineering",
+    "Topic :: Scientific/Engineering :: Information Analysis",
+    "Topic :: Scientific/Engineering :: Mathematics",
+    "Topic :: Scientific/Engineering :: Physics",
+    "Typing :: Typed",
+]
+dependencies = [
+    "importlib-metadata>=0.22;python_version<\"3.8\"",
+    "numpy>=1.13.3",
+    "packaging>=19.0",
+    "typing-extensions;python_version<\"3.8\"",
+]
+dynamic = [
+    "version",
+]
+
+[[project.authors]]
+name = "Jim Pivarski, Henry Schreiner, Eduardo Rodrigues"
+email = "eduardo.rodrigues@cern.ch"
+
+[project.optional-dependencies]
+all = [
+    "awkward>=1.2.0",
+    "ipykernel",
+    "myst-parser>0.13",
+    "nbsphinx",
+    "numba>=0.50; python_version>=\"3.6\"",
+    "pytest-cov>=3.0.0",
+    "pytest>=6",
+    "spark-parser",
+    "sphinx-math-dollar",
+    "Sphinx>=4.0",
+    "sphinx_book_theme>=0.0.42",
+    "sphinx_copybutton",
+    "uncompyle6",
+    "xdoctest>=1.0.0",
+]
+awkward = [
+    "awkward>=1.2.0",
+]
+dev = [
+    "awkward>=1.2.0",
+    "numba>=0.50; python_version>=\"3.6\"",
+    "pytest-cov>=3.0.0",
+    "pytest>=6",
+    "xdoctest>=1.0.0",
+]
+docs = [
+    "awkward",
+    "ipykernel",
+    "myst-parser>0.13",
+    "nbsphinx",
+    "sphinx-math-dollar",
+    "Sphinx>=4.0",
+    "sphinx_book_theme>=0.0.42",
+    "sphinx_copybutton",
+]
+test = [
+    "pytest-cov>=3.0.0",
+    "pytest>=6",
+    "xdoctest>=1.0.0",
+]
+test_extras = [
+    "spark-parser",
+    "uncompyle6",
+]
+
+[project.entry-points.numba_extensions]
+init = "vector:register_numba"
+
+[project.urls]
+"Bug Tracker" = "https://github.com/scikit-hep/vector/issues"
+Changelog = "https://vector.readthedocs.io/en/latest/changelog.html"
+Discussions = "https://github.com/scikit-hep/vector/discussions"
+Documentation = "https://vector.readthedocs.io/"
+Homepage = "https://github.com/scikit-hep/vector"
+
+[tool.hatch]
+version.source = "vcs"
+build.hooks.vcs.version-file = "src/vector/version.py"
+
+[tool.hatch.build.targets.sdist]
+include = [
+    "/src",
```

Comment:
Is this needed? It should be automatic, as long as the `.gitignore` is accurate, I think?